### PR TITLE
Add TLS and mTLS

### PR DESCRIPTION
**Pull request by jasonodonnell**
Currently the action does not support TLS or mTLS.  This adds new parameters to support these features:

| Parameter           | Default | Description                                                                                        |
|---------------------|---------|----------------------------------------------------------------------------------------------------|
| `caCertificate`     | null    | Base64 encoded CA certificate the server certificate was signed with.                              |
| `clientCertificate` | null    | Base64 encoded client certificate the action uses to authenticate with Vault when mTLS is enabled. |
| `clientKey`         | null    | Base64 encoded client key the action uses to authenticate with Vault when mTLS is enabled.         |
| `tlsSkipVerify`     | false   | When set to true, disables verification of server certificates when testing the action.            |

This current implementation expects that the client key is not password protected.  This could be added in the future if needed.

I made some other tweaks while I was in here:
- Expanded and organized `docker-compose.yaml` for the various tests
- Updated test formatting
- Added TLS testing infrastructure. The cert/keys added to this repo are valid for 100 years.
- Set secrets on the repo that contain the certificate/key tests will use.
- Updated `got` to 11.5.1.  This is required for the custom TLS settings.

All of these tests are run when a pull request is opened.  See the checks below for the specific test outputs.

### CODE REVIEW

File: src/action.js [hunk 1]
```diff
@@ -29,7 +29,28 @@ async function exportSecrets() {
 
     const defaultOptions = {
         prefixUrl: vaultUrl,
-        headers: {}
+        headers: {},
+        https: {}
+    }
+
+    const tlsSkipVerify = core.getInput('tlsSkipVerify', { required: false }) != 'false';
+    if (tlsSkipVerify == true) {
```
Comment (tomhjp) [inline src/action.js:37]: Is double equals here intentional? I think it allows type conversions before doing the comparison which is a bit ick. My understanding is most projects require triple equals equality checks everywhere for a modicum of type safety.

File: src/action.js [hunk 2]
```diff
@@ -29,7 +29,28 @@ async function exportSecrets() {
 
     const defaultOptions = {
         prefixUrl: vaultUrl,
-        headers: {}
+        headers: {},
+        https: {}
+    }
+
+    const tlsSkipVerify = core.getInput('tlsSkipVerify', { required: false }) != 'false';
```
Comment (tomhjp) [inline src/action.js:36]: What if someone sets this to `False`? Seems potentially a bit fragile if I'm correct that it's case-sensitive.

File: src/action.js (hunk 1 above)
Comment (jasonodonnell) [reply src/action.js]: Updated this to also check type for future proofing. 👍

File: src/action.js (hunk 2 above)
Comment (jasonodonnell) [reply src/action.js]: Casting to lower to make this a bit less fragile.

File: src/action.js (hunk 1 above)
Comment (tomhjp) [reply src/action.js]: Sorry I didn't make it explicit, but the same type-conversion applies to `!=` vs `!==` as well as `==` vs `===`, e.g. see https://stackoverflow.com/a/1946081

**Comment by RichiCoder1** [comment]
Whoop! Nice.

> I enjoy how deliciously go-like this JS is, lol. LGTM.

True to this action's history of being a mish-mash of language and pattern influences 😁

**Pull request merged**